ovirt/resource_ovirt_vm.go

```diff
@@ -938,9 +943,16 @@ func expandOvirtVMDiskAttachment(d interface{}, disk *ovirtsdk4.Disk) (*ovirtsdk
 	if disk != nil {
 		builder.Disk(disk)
 		if v, ok := dmap["size"]; ok {
-			newSize := int64(v.(int)) * int64(math.Pow(2, 30))
-			disk.SetProvisionedSize(newSize)
+			if v != 0 {
```

Comment:
Please remove the change for the provisionedSize. If there's no value the `ok` variable will be false.

Reply:
The condition `if v != 0 {` is necessary for this scenario:
```
resource "ovirt_vm" "vm" {
    ...
    template_id          = data.ovirt_templates.search_filtered_template.templates[0].id
    ...
    block_device {
        interface = var.vm_disk_interface
        alias = "vmname_Boot"
    }
}
```

When I need create new VM from template, I want to change only disk alias or interface type.
I don't need provision the new block_device.
By default, disk size return 0 if size parameter is not defined.

Therefore this check is necessary.
Without this check terraform will try setup disk size 0 for disk from template. Such attempts cause errors.